supervisor/homeassistant/module.py

```diff
@@ -481,7 +481,8 @@ def _restore_data():
                 ATTR_REFRESH_TOKEN,
                 ATTR_WATCHDOG,
             ):
-                self._data[attr] = data[attr]
+                if attr in data:
+                    self._data[attr] = data[attr]
```

Comment:
I'm aware this change feels unrelated to the PR. However after the changes this failed tests. Many of these attributes are `vol.Maybe` in the schema so they may not exist (and `refresh_token` in particular did not exist in our mock backups). This is an example of an error we were hiding. Nothing was raised and all we did was log an error so the test appeared to succeed, now it needs to be fixed.